src/pywifes/wifes_wsol.py

```diff
@@ -1570,58 +1562,130 @@ def derive_wifes_optical_wave_solution(inimg,
                                   for line in f1.readlines()])
       f1.close()
   # step 2 - find lines!
+  detsec = f[s].header['DETSEC']
+  y0 = int(detsec.split(',')[1].split(':')[0])
+  y1 = int(detsec.split(',')[1].split(':')[1].split(']')[0])
+  # Make the values work nicely with the range() command
+  if (y0 > y1):
+      ystop = y0 + 1
+      ystart = y1
+  else:
+      ystop = y1 + 1
+      ystart = y0
+  # Plot them or not ?
+  if doplot == True or ((doplot != False) and ('step1' in doplot)):
+      step1plot = True
+  else:
+      step1plot = False
+  if doplot == True or (doplot != False) and (('step2' in doplot)):
+      step2plot = True
+      return_dict['step2plot'] = step2plot
+  else:
+      step2plot = False    
+      return_dict['step2plot'] = step2plot
+
+  # guess the reference wavelengths
+  new_x, new_y, new_r = find_lines_and_guess_refs(
+      f[s].data,
+      s,
+      grating,
+      arc_name,
+      find_method=find_method,
+      shift_method=shift_method,
+      ref_arclines=ref_arclines,
+      dlam_cut_start=dlam_cut_start,
+      bin_x=bin_x, bin_y=bin_y,
+      yzp=ystart,
+      verbose=verbose,
+      flux_threshold_nsig=flux_threshold_nsig,
+      deriv_threshold_nsig=1.0,
+      multithread=multithread,
+      max_processes=max_processes,

```

Comment:
As in the comment on line 1522-1524 (can't link until the review is posted), I'd just set
```
multithread=False,
plot=step1plot
```
Skipping max_processes, and just setting multithread directly. 

You could add a comment above that parameter just in case someone looks at it and forgets the context that `run_slice` is singlethreaded.